xchange-stream-cexio/src/main/java/info/bitrich/xchangestream/cexio/CexioStreamingMarketDataService.java

```diff
@@ -29,17 +30,21 @@ public CexioStreamingMarketDataService(CexioStreamingRawService streamingOrderDa
           CexioWebSocketOrderBookSubscribeResponse, OrderBook> {
     BigInteger prevID = null;
     OrderBook orderBookSoFar =
-        new OrderBook(new Date(), new ArrayList<LimitOrder>(), new ArrayList<LimitOrder>());
+        new OrderBook(new Date(), new ArrayList<>(), new ArrayList<>());
     final CexioStreamingRawService streamingOrderDataService;
 
     public OrderBookUpdateConsumer(CexioStreamingRawService streamingOrderDataService) {
       this.streamingOrderDataService = streamingOrderDataService;
     }
 
+    @SuppressWarnings("RedundantThrows")
     @Override
-    public OrderBook apply(CexioWebSocketOrderBookSubscribeResponse t) throws Exception {
+    public OrderBook apply(@NonNull CexioWebSocketOrderBookSubscribeResponse t) throws Exception {
       OrderBook retVal;
       if (prevID != null && prevID.add(BigInteger.ONE).compareTo(t.id) != 0) {
+        prevID = null;
+        orderBookSoFar = new OrderBook(new Date(), new ArrayList<>(), new ArrayList<>());
```

Comment:
is there some base layer of the orderbook that should be applied like a snapshot?

Reply:
Good question. The documentation doesn't mention it but it seems like it might be a good idea. I can take a closer look and find out.